presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetastoreClosure.java

```diff
@@ -67,12 +77,14 @@ public HiveMetastoreClosure(HiveMetastore delegate)
 
     public PartitionStatistics getTableStatistics(HiveIdentity identity, String databaseName, String tableName)
     {
-        return delegate.getTableStatistics(identity, databaseName, tableName);
+        return delegate.getTableStatistics(identity, getExistingTable(identity, databaseName, tableName));
     }
 
     public Map<String, PartitionStatistics> getPartitionStatistics(HiveIdentity identity, String databaseName, String tableName, Set<String> partitionNames)
     {
-        return delegate.getPartitionStatistics(identity, databaseName, tableName, partitionNames);
+        Table table = getExistingTable(identity, databaseName, tableName);
+        List<Partition> partitions = getExistingPartitionsByNames(identity, table, ImmutableList.copyOf(partitionNames));
```

Comment:
Use `Set<String> partitionNames` in `getExistingPartitionsByNames` to avoid copying

Reply:
`HiveMetastore#getPartitionsByNames` already accepts list of partition names and I didn't want to change that. That list goes down to thrift level